Mobile/ShopLens/Droid/Source/AppPreferences.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using Android.Content;
+
+namespace ShopLens.Droid.Source
+{
+    public class ActivityPreferences
+    {
+        ISharedPreferences prefs;
+        ISharedPreferencesEditor prefsEditor;
+        Context pContext;
+        List<string> items;
+        int itemCounter = 0;
+
+        public ActivityPreferences(Context context, string name)
+        {
+            pContext = context;
+            prefs = pContext.GetSharedPreferences(name, FileCreationMode.Private);
+            prefsEditor = prefs.Edit();
+            foreach (KeyValuePair<string, object> entry in prefs.All)
+            {
+                itemCounter += 1;
+            }
+        }
+
+        public List<string> GetPreferencesToList()
+        {
+            if (items == null)
+            {
+                items = new List<string> { };
+            }
+            foreach (KeyValuePair<string, object> entry in prefs.All)
+            {
+                items.Add(entry.Value.ToString());
```

Comment:
Maybe these lines (27 - 34) can be done with LINQ? Not that it's bad, but just as LINQ practice for the exam

Reply:
Noted for later.